[Common] fix tensor dimension compare util

niley7464 opened the pull request:
This patch fixes `gst_tensor_dimension_is_equal`
and adds unittests.

**Self evaluation:**
1. Build test: [*]Passed [ ]Failed [ ]Skipped
2. Run test: [*]Passed [ ]Failed [ ]Skipped

myungjoo commented on gst/nnstreamer/nnstreamer_plugin_api_util_impl.c:
```diff
@@ -1027,9 +1027,9 @@ gst_tensor_dimension_is_equal (const tensor_dim dim1, const tensor_dim dim2)
+        remained = (const tensor_dim *) dim2;
       else if (dim2[i] == 0U)
-        remained = (const tensor_dim *) &dim1;
+        remained = (const tensor_dim *) dim1;
```
```const tensor_dim *remained```
and
```const tensor_dim dim1, dim2```.
Not safe.

niley7464 commented:
> 1. Be prepared for erroneous tensor ( 3: 4: 0: 0: 2: 0: ...).
> 2. Are you really sure with `remained = (const tensor_dim *) dim1;`?

Removed `remained` tensor dimension and uses `gst_tensor_dimension_is_valid`.
Also, I added some testcase for erroneous tensor.
Thank you 👍
